packages/core/src/hooks/useTokenList.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react'
+import axios from 'axios'
+import { TokenInfo } from '@uniswap/token-lists'
+import { ChainId } from '../constants'
+
+interface TokenList {
+  name: string
+  logoURI: string
+  tokens: TokenInfo[]
+}
+
+export function useTokenList(tokenListURI: string, chainId: ChainId = ChainId.Mainnet) {
+  const [tokenList, setTokenList] = useState<TokenList>()
+
+  useEffect(() => {
+    axios
+      .get(tokenListURI)
+      .then((response) => {
+        const { name, logoURI, tokens } = response.data
+        setTokenList({
+          name,
+          logoURI,
+          tokens: (tokens as TokenInfo[]).filter((token) => token.chainId === chainId),
+        })
+      })
+      .catch((err) => {
+        console.log(err)
+        setTokenList(undefined)
+      })
+  }, [tokenListURI, chainId])
```

Comment:
Could just use plain fetch with checking if response is ok 
```js
.fetch(url)
    .then(async response => {
      if (response.ok) {
        return await response.json()
      } else {
        const errorMessage = await response.text()
        return Promise.reject(new Error(errorMessage))
      }
    }).catch(err)...
```

Reply:
✅